libs/internal/src/serialization/json_flag.cpp

```diff
@@ -208,4 +208,160 @@ tag_invoke(boost::json::value_to_tag<
     return std::make_optional(variation);
 }
 
+namespace data_model {
+void tag_invoke(
+    boost::json::value_from_tag const& unused,
+    boost::json::value& json_value,
+    data_model::Flag::Rollout::WeightedVariation const& weighted_variation) {
+    auto& obj = json_value.emplace_object();
+    obj.emplace("variation", weighted_variation.variation);
+    obj.emplace("weight", weighted_variation.weight);
+
+    WriteMinimal(obj, "untracked", weighted_variation.untracked);
+}
+
+void tag_invoke(boost::json::value_from_tag const& unused,
+                boost::json::value& json_value,
+                data_model::Flag::Rollout::Kind const& kind) {
+    switch (kind) {
+        case Flag::Rollout::Kind::kUnrecognized:
+            // TODO: Should we be preserving the original string.
+            break;
+        case Flag::Rollout::Kind::kExperiment:
+            json_value.emplace_string() = "experiment";
+            break;
+        case Flag::Rollout::Kind::kRollout:
+            json_value.emplace_string() = "rollout";
+            break;
+    }
+}
+
+void tag_invoke(boost::json::value_from_tag const& unused,
+                boost::json::value& json_value,
+                data_model::Flag::Rollout const& rollout) {
+    auto& obj = json_value.emplace_object();
+
+    obj.emplace("variations", boost::json::value_from(rollout.variations));
+    if (rollout.kind != Flag::Rollout::Kind::kUnrecognized) {
+        // TODO: Should we be preserving the original string and putting it in.
+        obj.emplace("kind", boost::json::value_from(rollout.kind));
+    }
+    WriteMinimal(obj, "seed", rollout.seed);
+    if (rollout.bucketBy.Valid()) {
+        obj.emplace("bucketBy", rollout.bucketBy.RedactionName());
```

Comment:
q: Should be preserving this even if invalid?

Reply:
Maybe?

Reply:
I am fine with preserving it, because here we will have some logical value retained. Probably.